src/extension.ts

```diff
@@ -304,7 +304,7 @@ class CodeProvider implements vscode.TextDocumentContentProvider {
                 this._documentText,
                 this._inputKind,
                 this._targetLanguage,
-                "TopLevel",
+                path.basename(this._document.fileName, ".json"),
```

Comment:
We should also check for extensions `.schema` and `.ts`, since those can be inputs, too.